MuscleModel/ReadFromModel/getMuscleInfo.m

```diff
@@ -1,154 +1,162 @@
+    % Pre-allocate loop variables
+    DOF_inds=nan(length(Misc.DofNames_Input{t}),1);
+    ct=1;
+
+    % Loop over each DOF in the model
+    for i=1:length(Misc.DofNames_Input{t})
+        % read the Muscle Analysis Result
+        MA_FileName=fullfile(Misc.MuscleAnalysisPath,[Misc.MAtrialName{t} '_MuscleAnalysis_MomentArm_' Misc.DofNames_Input{t}{i} '.sto']);
+        if exist(MA_FileName,'file')
+            dm_Data_temp=importdata(fullfile(Misc.MuscleAnalysisPath,[Misc.MAtrialName{t} '_MuscleAnalysis_MomentArm_' Misc.DofNames_Input{t}{i} '.sto']));
+        else
+            error(['Cannot open muscle analysis results for: ' Misc.DofNames_Input{t}{i}])
+        end
+
+        % get the indexes for the selected MuscleNames (only needed in first iteration)
+        if i==1
+            nfr = length(dm_Data_temp.data(:,1));
+            headers=dm_Data_temp.colheaders;
+            Inds_muscles=nan(length(Misc.MuscleNames_Input{t}),1);
+            IndsNames_sel=nan(length(Misc.MuscleNames_Input{t}),1);
+            ctm=1;
+            for j=1:length(Misc.MuscleNames_Input{t})
+                ind_sel=find(strcmp(Misc.MuscleNames_Input{t}{j},headers));
+                if ~isempty(ind_sel)
+                    Inds_muscles(ctm)=ind_sel; IndsNames_sel(ctm)=j;
+                    ctm=ctm+1;
+                else
+                    disp(['Warning: The selected muscle ' Misc.MuscleNames_Input{t}{j} ' does not exist in the selected model of trial' Misc.trialName{t} '. This muscles is removed from the program']);
+                end
             end
+            Misc.MuscleNames{t}=Misc.MuscleNames_Input{t}(IndsNames_sel);
+            Misc.idx_allMuscleList{t}=Misc.idx_allMuscleList_Input{t}(IndsNames_sel);
+            Inds_muscles(isnan(Inds_muscles))=[];                               % Delete the muscles names that are not selected by the user
+            dM_temp=nan(nfr,length(Misc.DofNames_Input{t}),length(Misc.MuscleNames{t}));    % pre-allocate moment arms
         end
-        Misc.MuscleNames=Misc.MuscleNames_Input(IndsNames_sel);
-        Inds_muscles(isnan(Inds_muscles))=[];                               % Delete the muscles names that are not selected by the user        
-        dM_temp=nan(nfr,length(Misc.DofNames_Input),length(Misc.MuscleNames));    % pre-allocate moment arms
+
+        % Evaluate if one of the muscles spans this DOF (when moment arms > 0.001)
+        dM=dm_Data_temp.data(:,Inds_muscles);    
+        if any(any(abs(dM)>0.001))
+            Misc.DofNames_muscles{t}{ct}=Misc.DofNames_Input{t}{i};
+            dM_temp(:,i,:)=dM;
+            DOF_inds(ct)=i;
+            ct=ct+1;   
+        end     
     end
-    
-    % Evaluate if one of the muscles spans this DOF (when moment arms > 0.001)
-    dM=dm_Data_temp.data(:,Inds_muscles);    
-    if any(any(abs(dM)>0.001))
```

Comment:
we probably want to decrease this treshold. Maybe 0.0001 ?

Reply:
done